pkg/server/handler.go

```diff
@@ -225,6 +226,9 @@ func WithWorkspaceProjection(apiHandler http.Handler) http.HandlerFunc {
 			newPath := path.Join("/services/workspaces", cluster.Name.String(), req.URL.Path)
 			klog.V(4).Infof("Rewriting %s -> %s", path.Join(cluster.Name.Path(), req.URL.Path), newPath)
 			req.URL.Path = newPath
+			if !opts.Virtual.Enabled && opts.Extra.ShardVirtualWorkspaceURL != "" {
```

Comment:
since we removed the `installVirtualWorkspacesRedirect` method, should we rewrite here?

Reply:
Done in rebase code.